device/create.go

```diff
@@ -0,0 +1,69 @@
+package device
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/jinzhu/copier"
+	"github.com/traggo/server/auth"
+	"github.com/traggo/server/auth/rand"
+	"github.com/traggo/server/generated/gqlmodel"
+	"github.com/traggo/server/model"
+	"github.com/traggo/server/user/password"
+)
+
+var (
+	timeNow          = time.Now
+	randToken        = rand.Token
+	comparePassword  = password.ComparePassword
+	errUserPassWrong = errors.New("username/password combination does not exist")
+)
+
+// CreateDevice creates a device.
+func (r *ResolverForDevice) CreateDevice(ctx context.Context, username string, pass string, deviceName string, expiresAt model.Time, cookie bool) (*gqlmodel.Login, error) {
+	if !expiresAt.Time().After(timeNow()) {
+		return nil, errors.New("expiresAt must be in the future")
+	}
+
+	user := new(model.User)
+	find := r.DB.Where("name = ?", username).Find(user)
+
+	if find.RecordNotFound() {
+		return nil, errUserPassWrong
+	}
+
+	if !comparePassword(user.Pass, []byte(pass)) {
+		return nil, errUserPassWrong
+	}
+
+	token := randToken(20)
+	for !r.DB.Where("token = ?", token).Find(new(model.Device)).RecordNotFound() {
+		token = randToken(20)
+	}
```

Comment:
~I think an explanation what happens here...~
~`RecordNotFound()` sounds more like a boolean and `for` sound like an iteration.~

Got it! `for` is like a while in this case, and we want to avoid the usage of the same token again ^^